### PR TITLE
Switch to joi

### PR DESCRIPTION
**Pull request by jessehansen**
I'm open to suggestions for how to better use this while staying backwards-compatible. It ended up looking mostly the same as the assert-plus validation.

### CODE REVIEW

File: src/schemas/trusted-client-schema.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  log: joi.object()
+    .keys({
+      error: joi.func().required(),
+      warn: joi.func().required(),
+      info: joi.func().required(),
+      debug: joi.func().required()
+    })
+    .unknown(true)
+    .description('Optional logger implementation')
+    .optional()
```
Comment (rizowski) [inline src/schemas/trusted-client-schema.js]: For Joi, optional properties are implied. So you only need to specify the required properties via `.required()`

**Comment by rizowski** [comment]
Only one comment on the schema's themselves but other than that 👍

**Comment by jessehansen** [comment]
fixed

**Comment by rizowski** [comment]
Looks good.

**Pull request merged**